syntax = "proto3";

package plosolver.core;

option go_package = "github.com/plosolver/backend/protos/core";
option java_multiple_files = true;
option java_package = "com.plosolver.grpc.core";
option java_outer_classname = "CoreProto";

import "google/protobuf/timestamp.proto";
import "google/protobuf/empty.proto";
import "common.proto";

// =============================================================================
// Core Service
// =============================================================================

service CoreService {
  rpc HealthCheck(google.protobuf.Empty) returns (HealthCheckResponse);
  rpc GetSystemStatus(google.protobuf.Empty) returns (SystemStatusResponse);
  rpc GetUserStats(GetUserStatsRequest) returns (GetUserStatsResponse);
}

message HealthCheckResponse {
  string status = 1;
  string version = 2;
  google.protobuf.Timestamp timestamp = 3;
}

message SystemStatusResponse {
  string status = 1;
  map<string, string> services = 2;
  map<string, double> metrics = 3;
  plosolver.common.Error error = 4;
}

message GetUserStatsRequest {
  string user_id = 1;
}

message GetUserStatsResponse {
  UserStats stats = 1;
  plosolver.common.Error error = 2;
}

message UserStats {
  string user_id = 1;
  int32 total_analyses = 2;
  int32 analyses_this_month = 3;
  int32 credits_used = 4;
  int32 credits_remaining = 5;
  google.protobuf.Timestamp last_analysis = 6;
  google.protobuf.Timestamp member_since = 7;
}
